plugins/woocommerce/includes/admin/class-wc-admin-notices.php

```diff
@@ -115,6 +119,46 @@ public static function reset_admin_notices() {
 		self::add_notice( 'template_files' );
 		self::add_min_version_notice();
 		self::add_maxmind_missing_license_key_notice();
+		self::maybe_add_legacy_api_removal_notice();
+	}
+
+	/**
+	 * Add an admin notice about the removal of the Legacy REST API if the said API is enabled.
+	 *
+	 * TODO: Remove this method in WooCommerce 9.0.
+	 */
+	private static function maybe_add_legacy_api_removal_notice() {
+		if ( 'yes' !== get_option( 'woocommerce_api_enabled' ) ) {
+			return;
+		}
+
+		self::add_custom_notice(
+			'legacy_api_removed_in_woo_90',
+			sprintf(
+				'%s%s',
+				sprintf(
+					'<h4>%s</h4>',
+					esc_html__( 'WooCommerce Legacy API to be removed soon', 'woocommerce' )
```

Comment:
```suggestion
					esc_html__( 'The Legacy REST API will be removed soon', 'woocommerce' )
```

Reply:
I added the word "WooCommerce" on purpose because the notice can appear outside the WooCommerce admin context (for example it appears in the plugins page), similar to the "WooCommerce database update required" message.

Reply:
Ah, that makes sense.